### PR TITLE
add build cache

### PR DESCRIPTION
**Pull request by chenshuai2144**
Improve the compilation speed by comparing hashes to cache

### CODE REVIEW

**Comment by chenshuai2144** [comment]
Can you give me permission for npm and github?？

**Comment by mzohaibqc** [comment]
@chenshuai2144 can you spare some time to discuss these changes with me on google hangout. mzohaib.qc@gmail.com is my email.

 I'll allow you write access.

**Comment by chenshuai2144** [comment]
Can you review this pr ？

File: package.json [hunk 1]
```diff
@@ -1,18 +1,24 @@
 {
-  "name": "antd-theme-generator",
-  "version": "1.0.7",
-  "description":
-    "This a script to generate color specific styles less file and which you can use to change theme dynamically in browser",
+  "name": "antd-pro-theme-generator",
```
Comment (mzohaibqc) [inline package.json]: it's  `antd-theme-generator` and not `ant-pro-theme-generator`.

File: package.json [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+  "version": "1.0.0",
```
Comment (mzohaibqc) [inline package.json]: @chenshuai2144 Is it intentional ? it should be "1.0.8"

File: package.json [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+  "description": "This a script to generate color specific styles less file and which you can use to change theme dynamically in browser",
```
Comment (mzohaibqc) [inline package.json]: `"This is a script to generate ....` error in orignal description, can you change it as part of your changes

File: index.js [hunk 4]
```diff
@@ -182,8 +186,14 @@ function getLessVars(filtPath) {
 */
 function getShade(varName) {
   let [, className, number] = varName.match(/(.*)-(\d)/);
-  if (/primary-\d/.test(varName)) className = '@primary-color';
-  return 'color(~`colorPalette("@{' + className.replace('@', '') + '}", ' + number + ")`)";
+  if (/primary-\d/.test(varName)) className = "@primary-color";
```
Comment (mzohaibqc) [inline index.js]: @chenshuai2144 IMO, single quotes are preffered.

**Pull request closed without merging**